be/src/exprs/vectorized/json_functions.cpp

```diff
@@ -337,27 +353,255 @@ ColumnPtr JsonFunctions::get_json_string(FunctionContext* context, const Columns
+}
+
+ColumnPtr JsonFunctions::json_exists(FunctionContext* context, const Columns& columns) {
+    auto num_rows = columns[0]->size();
+    auto json_viewer = ColumnViewer<TYPE_JSON>(columns[0]);
+    auto path_viewer = ColumnViewer<TYPE_VARCHAR>(columns[1]);
+    ColumnBuilder<TYPE_BOOLEAN> result(num_rows);
+
+    JsonPath stored_path;
+    for (int row = 0; row < num_rows; row++) {
+        if (json_viewer.is_null(row) || json_viewer.value(row) == nullptr) {
+            result.append_null();
+            continue;
+        }
+
+        JsonValue* json_value = json_viewer.value(row);
+        Slice path_str = path_viewer.value(row);
+        auto jsonpath = get_prepared_or_parse(context, path_str, &stored_path);
+
+        if (!jsonpath.ok()) {
+            result.append_null();
+            VLOG(2) << "parse json path failed: " << path_str;
+            continue;
+        }
+        VLOG(2) << "json_exists for  " << path_str << " of " << json_value->to_string().value();
+        vpack::Builder builder;
+        vpack::Slice slice = JsonPath::extract(json_value, *jsonpath.value(), &builder);
+        result.append(!slice.isNone());
+    }
+
+    return result.build(ColumnHelper::is_all_const(columns));
 }
 
 ColumnPtr JsonFunctions::json_array(FunctionContext* context, const Columns& columns) {
-    // TODO(mofei) implement
-    return fake_result(columns[0]->size());
+    namespace vpack = arangodb::velocypack;
+
+    // empty arguments create an empty json array
+    if (columns.size() == 0) {
+        ColumnBuilder<TYPE_JSON> result(1);
+        vpack::Builder builder;
+        builder.openArray();
+        builder.close();
+        vpack::Slice json_slice = builder.slice();
+        JsonValue json(json_slice);
+        result.append(&json);
+        return result.build(ColumnHelper::is_all_const(columns));
+    }
+
+    size_t rows = columns[0]->size();
+    ColumnBuilder<TYPE_JSON> result(rows);
+    std::vector<ColumnViewer<TYPE_JSON>> viewers;
+    for (auto& col : columns) {
+        viewers.emplace_back(ColumnViewer<TYPE_JSON>(col));
+    }
+
+    for (int row = 0; row < rows; row++) {
+        vpack::Builder builder;
+        {
+            vpack::ArrayBuilder ab(&builder);
+            for (auto& view : viewers) {
+                if (view.is_null(row)) {
+                    builder.add(vpack::Value(vpack::ValueType::Null));
+                } else {
+                    JsonValue* json = view.value(row);
+                    builder.add(json->to_vslice());
+                }
+            }
+        }
+        vpack::Slice json_slice = builder.slice();
+        JsonValue json(json_slice);
+        result.append(&json);
+    }
+    return result.build(ColumnHelper::is_all_const(columns));
+}
+
+ColumnPtr JsonFunctions::json_object(FunctionContext* context, const Columns& columns) {
+    namespace vpack = arangodb::velocypack;
+
+    // empty arguments create an empty object
+    if (columns.size() == 0) {
```

Comment:
This process way seems has an issue. please test it with real data set. The result column size shouldn't always 1.

Reply:
Fixed. 

Actually this function could not be matched with empty arguments, and throws an error `No matching function with signature: json_array`.

So I another function signature `json_object()` with empty arguments, and another function implementations.